code/modules/atmospherics/machinery/components/binary_devices/valve.dm

```diff
@@ -42,6 +42,7 @@ It's like a regular ol' straight pipe, but you can turn it on and off.
 		parent1.reconcile_air()
 		investigate_log("was opened by [usr ? key_name(usr) : "a remote signal"]", INVESTIGATE_ATMOS)
 		vent_movement |= VENTCRAWL_ALLOWED
+	SEND_SIGNAL(src, COMSIG_VALVE_TOGGLED, on)
```

Comment:
Similar logic as before, try to put your signals at the start of the proc whenever possible, before the change happens.